src/openforms/analytics_tools/tests/test_models.py

```diff
@@ -81,4 +170,279 @@ def test_analytics_tool_enabled_but_related_fields_are_not(self):
 
         for csp in self.json_csp:
             with self.subTest("Test deletion of CSP"):
-                self.assertFalse(CSPSetting.objects.filter(value=csp["value"]).exists())
+                self.assertFalse(
+                    CSPSetting.objects.filter(
+                        value=csp["value"], directive=csp["directive"]
+                    ).exists()
+                )
+
+    def test_matomo_enabled_but_related_fields_are_not(self):
+        self.config.enable_matomo_site_analytics = True
+        self.config.matomo_url = ""
+
+        with self.assertRaises(ValidationError):
+            self.config.clean()
+
+
+@override_settings(
+    SOLO_CACHE=None,
+)
+class PiwikTests(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+        cls.piwik_url = "https://example.com"
+        cls.piwik_site_id = 1234
+        string_replacements_list = [
+            ("SITE_ID", cls.piwik_site_id),
+            ("SITE_URL", cls.piwik_url),
+            (
+                "DOMAIN_HASH",
+                lambda cookie: get_domain_hash(
+                    settings.ALLOWED_HOSTS[0], cookie_path=cookie["path"]
+                ),
+            ),
+        ]
+        cls.json_cookies = get_cookies(
+            analytics_tool=AnalyticsTools.piwik,
+            string_replacement_list=string_replacements_list,
+        )
+
+        cls.json_csp = get_csp(
+            analytics_tool=AnalyticsTools.piwik,
+            string_replacement_list=string_replacements_list,
+        )
+
+    def setUp(self):
+        super().setUp()
+        self.config = AnalyticsToolsConfiguration.get_solo()
+        self.config.analytics_cookie_consent_group = CookieGroup.objects.get(
+            varname="analytical"
+        )
+        self.config.piwik_url = self.piwik_url
+        self.config.piwik_site_id = self.piwik_site_id
+        self.addCleanup(lambda: AnalyticsToolsConfiguration.get_solo().delete())
```

Comment:
this is the "not-interesting" part that you could put in a mixin or base test case class, used by all the tool-specific test cases. That would cut down a bit on setup code.

I think I'd also split up the test file into a test file per analytics provider.